operations/patch_util.go

```diff
@@ -354,56 +354,102 @@ func getPatchDisplay(p *patch.Patch, summarize bool, uiHost string) (string, err
 	return out.String(), nil
 }
 
+func isCommitRange(commits string) bool {
+	return strings.Contains(commits, "..")
+}
+
+func formatCommitRange(commits string) string {
+	if commits == "" {
+		return commits
+	}
+	if isCommitRange(commits) {
+		return commits
+	}
+	return fmt.Sprintf("%s^!", commits)
+}
+
+func getFeatureBranch(ref, commits string) string {
+	if ref != "" {
+		return ref
+	}
+	if commits != "" {
+		// if one commit, this returns just that commit, else the first commit in the range
+		return strings.Split(commits, "..")[0]
+	}
+	return "HEAD"
+}
+
+func isValidCommitsFormat(commits string) error {
+	errToReturn := errors.New("Invalid commit format: verify input is of the form `<hash1> OR `<hash1>..<hash2>` (where hash1 is an ancestor of hash2)")
+	if commits == "" || !isCommitRange(commits) {
+		return nil
+	}
+
+	commitsList := strings.Split(commits, "..")
+	if len(commitsList) != 2 { // extra check
+		return errToReturn
+	}
+
+	for i := range commitsList {
+		commitsList[i] = strings.Trim(commitsList[i], ".") // handle extra dot notation
```

Comment:
won't the extra "." just be on the second commit?

Reply:
Yeah theoretically, unless like they have four "..." or something but I figured it doesn't hurt to be safe.